SDKBuildScripts/mac_unity_RunAutoTests.sh

```diff
@@ -0,0 +1,155 @@
+    SetEachProjDefine ${SdkName}_TS
+    SetEachProjDefine ${SdkName}_TZ
+}
+
+SetEachProjDefine() {
+    echo === Set Each Proj Define ===
+    pushd "${ProjRootPath}/$1"
+    $UNITY191 -buildOSXUniversalPlayer "${ProjRootPath}/${SdkName}_TC/" -accept-apiupdate -disable-assembly-updater -noUpm -nographics -quit -batchmode -executeMethod SetupPlayFabExample.Setup -logFile "${ProjRootPath}/compile$1.txt" || (cat "${ProjRootPath}/compile$1.txt" && return 1)
+    popd
+}
+
+JenkernaughtSaveCloudScriptResults() {
+    echo === Save test results to Jenkernaught ===
+    pushd "$WORKSPACE/SDKGenerator/JenkinsConsoleUtility/bin/Debug"
+    # cmd <<< "JenkinsConsoleUtility --listencs -buildIdentifier $BuildIdentifier -workspacePath $WORKSPACE -timeout 30 -verbose true"
+    . ./JenkinsConsoleUtility --listencs -buildIdentifier $BuildIdentifier -workspacePath $WORKSPACE -timeout 30 -verbose true
+    popd
+}
+
+RunClientJenkernaught() {
+    echo === Build OSX Client Target ===
+    pushd "${ProjRootPath}/${SdkName}_TC/"
+    $UNITY191 -buildOSXUniversalPlayer "${ProjRootPath}/${SdkName}_TC" -accept-apiupdate -disable-assembly-updater -noUpm -nographics -quit -batchmode -executeMethod PlayFab.Internal.PlayFabPackager.MakeOsxBuild -logFile "${ProjRootPath}/buildOSXClient.txt" || (cat "${ProjRootPath}/buildOSXClient.txt" && return 1)
+    popd
+
+    echo === Run the $UNITY183 Client UnitTests ===
+    pushd "${ProjRootPath}/${SdkName}_TC/testBuilds"
+    ls
+    popd
+
+    JenkernaughtSaveCloudScriptResults
+    if [[ $? -ne 0 ]]; then return 1; fi
+}
+
+BuildClientByFunc() {
+    if [ "$1" = "true" ]; then
+        echo === Build $2 Target ===
+        pushd "${ProjRootPath}/${SdkName}_TC/"
+        $UNITY191 -buildOSX64Player "${ProjRootPath}/${SdkName}_TC" -quit -batchmode -executeMethod PlayFab.Internal.PlayFabPackager.$2 -logFile "${ProjRootPath}/${2}.txt" || (cat "${ProjRootPath}/${2}.txt" && return 1)
+        popd
+        #Run the console test command if present
+        if [ ! -z "$3" ]; then
+            $3
+        fi
+    fi
+}
+
+BuildMainPackage() {
+    if [ "$UNITY_PUBLISH_VERSION" = "$UNITY_VERSION" ] && [ "$BuildMainUnityPackage" = "true" ]; then
+        echo === Build the asset bundle ===
+        cd "$WORKSPACE/$UNITY_VERSION/${SdkName}_BUP"
+        $UNITY183 -projectPath "$WORKSPACE/$UNITY_VERSION/${SdkName}_BUP" -quit -batchmode -executeMethod PlayFab.Internal.PlayFabPackager.PackagePlayFabSdk -logFile "$WORKSPACE/${SdkName}/buildPackageOutput.txt" || (cat "$WORKSPACE/${SdkName}/buildPackageOutput.txt" && return 1)
+    fi
+}
+
+#pretty-print test a  test result codes (if the test was actually run)
+EM() {
+    if [ "$2" != "true" ]; then echo "N/A"
+    elif [ $1 -ne 0 ]; then echo FAIL
+    else echo PASS
+    fi
+}
+
+#boil error codes down to either 1 or 0.
+EC() {
+    if [ $1 -ne 0 ]; then return 1; else return 0; fi
+}
+
+DoWork() {
+    CheckVars
+    SetProjDefines
+    RunClientJenkernaught
```

Comment:
Are we actually checking for cloudscript results in this case?  I thought the intention was to simply check compilation?  If we are meaning to check the test results, I didn't see any of the normal output I'd expect to see in the jenkins console output here: https://pf-jenkins.playfab.com/view/SDK/job/SDK_Work_Unity_MacOS/5/console